### PR TITLE
1.sqlserver and sqlserver ro instance add tags. 2. es support disksiz…

### PR DESCRIPTION
**Pull request by cyberHermanwang**
1.sqlserver and sqlserver ro instance add tags. 
2. es support disksize and nodeType modification.

### CODE REVIEW

**Comment by cyberHermanwang** [comment]
make testacc TESTARGS="-run=TestAccTencentCloudElasticsearchInstance_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestAccTencentCloudElasticsearchInstance_basic -timeout 120m
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/tencentcloudstack/terraform-provider-tencentcloud/gendoc	(cached) [no tests to run]
=== RUN   TestAccTencentCloudElasticsearchInstance_basic
=== PAUSE TestAccTencentCloudElasticsearchInstance_basic
=== CONT  TestAccTencentCloudElasticsearchInstance_basic
--- PASS: TestAccTencentCloudElasticsearchInstance_basic (1178.37s)
PASS
ok  	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud	1179.646s
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/connectivity	[no test files]
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/internal/helper	[no test files]
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/ratelimit	[no test files]

File: tencentcloud/resource_tc_elasticsearch_instance.go [hunk 1]
```diff
@@ -562,6 +561,52 @@ func resourceTencentCloudElasticsearchInstanceUpdate(d *schema.ResourceData, met
...
+			if errRet != nil {
+				return retryError(errRet)
+			}
+			return nil
+		})
+		if err != nil {
+			return err
+		}
+		d.SetPartial("node_info_list")
+		err = resource.Retry(10*readRetryTimeout, func() *resource.RetryError {
```
Comment (gailwang) [inline tencentcloud/resource_tc_elasticsearch_instance.go:596]: update使用 writeRetryTimeout

File: tencentcloud/resource_tc_elasticsearch_instance.go [hunk 2]
```diff
@@ -562,6 +561,52 @@ func resourceTencentCloudElasticsearchInstanceUpdate(d *schema.ResourceData, met
 		d.SetPartial("basic_security_type")
 	}
 
+	if v, ok := d.GetOk("node_info_list"); ok {
```
Comment (gailwang) [inline tencentcloud/resource_tc_elasticsearch_instance.go:564]: 这里update不是应该判断hasChange么

File: tencentcloud/resource_tc_elasticsearch_instance.go (hunk 1 above)
Comment (cyberHermanwang) [reply tencentcloud/resource_tc_elasticsearch_instance.go:596]: 这里是 Describe接口，不是update接口。

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
 ## 1.46.5 (Unreleased)
 
+ENHANCEMENTS:
+
+* Resource: `tencentcloud_sqlserver_instance` add new argument `tags`.
+* Resource: `tencentcloud_sqlserver_readonly_instance` add new argument `tags`.
+* Resource: `tencentcloud_elasticsearch_instance` support nodeType and diskSize modification.
+
```
Comment (gailwang) [inline CHANGELOG.md:8]: DataSource的tag log信息没写

**Comment by cyberHermanwang** [comment]
make testacc TESTARGS="-run=TestAccTencentCloudElasticsearchInstance_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestAccTencentCloudElasticsearchInstance_basic -timeout 120m
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/tencentcloudstack/terraform-provider-tencentcloud/gendoc	(cached) [no tests to run]
=== RUN   TestAccTencentCloudElasticsearchInstance_basic
=== PAUSE TestAccTencentCloudElasticsearchInstance_basic
=== CONT  TestAccTencentCloudElasticsearchInstance_basic
=========================================
--- PASS: TestAccTencentCloudElasticsearchInstance_basic (1742.97s)
PASS
ok  	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud	1743.994s
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/connectivity	[no test files]
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/internal/helper	[no test files]
?   	github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/ratelimit	[no test files]

**Review by gailwang — APPROVED** [review]

**Review by crab21 — APPROVED** [review]

**Review by ChuGuai — APPROVED** [review]

File: tencentcloud/resource_tc_elasticsearch_instance.go [hunk 4]
```diff
@@ -562,6 +561,54 @@ func resourceTencentCloudElasticsearchInstanceUpdate(d *schema.ResourceData, met
 		d.SetPartial("basic_security_type")
 	}
 
+	if d.HasChange("node_info_list") {
+		if v, ok := d.GetOk("node_info_list"); ok {
```
Comment (gailwang) [inline tencentcloud/resource_tc_elasticsearch_instance.go:565]: required参数不用判断

**Pull request merged**